adobe/react-spectrum
# fix(#2745): useNumberField ignores name prop

Updates the `useNumberField` hook to automatically pass props not explicitly listed in the function (e.g. `name` or `placeholder`) to `useFormattedTextField`, so they can be propagated to the `inputProps`..  As part of this, the `AriaNumberFieldProps` definition was updated to include properties from `TextInputDOMProps` (omitting the ones already defined in elsewhere in the interface, explicitly overwritten by `useNumberField`, or that aren't natively supported for number fields, like `pattern`).  

Also adds a test file for `useNumberField`, but only included tests for passing `name` and `placeholder` (as well as one additional one for no props), as well as 3 new stories for the `NumberField` storybook story (one for `name`, one for `placeholder`, and a set for the additional input actions from `TextInputDOMProps` )

Closes [#2745](https://github.com/adobe/react-spectrum/issues/2745)

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue 2745](https://github.com/adobe/react-spectrum/issues/2745).

## 📝 Test Instructions:

1. Open the `NumberField` story in Storybook
2. Navigate to the `placeholder` story, "Enter value" should display in the number field
3. Navigate to the `with form name` story
4. Open browser dev tools and inspect the `NumberField` element (specifically the input field)
5. Element should have `name="numberfield-input-form-name"` 
6. Navigate to the `input events` story
7. Enter values into the number field, events should fire as data is entered, selected, cut, copied, or pasted

## 🧢 Your Project:

The Trade Desk, Inc.
